src/extension/features/general/pos-style-currency-entry-mode/index.js

```diff
@@ -0,0 +1,124 @@
+import { create, all } from 'mathjs';
+import { Feature } from 'toolkit/extension/features/feature';
+import { componentLookup } from 'toolkit/extension/utils/ember';
+
+export class POSStyleCurrencyEntryMode extends Feature {
+  constructor() {
+    super();
+
+    this.mathEvaluatorInstance = null;
+    this.currencyFormatter = null;
+    this.accountCurrency = null;
+    this.decimalDigits = null;
+  }
+
+  shouldInvoke() {
+    const {
+      currencyFormatter,
+    } = ynab.YNABSharedLibWebInstance.firstInstanceCreated.formattingManager;
+
+    this.currencyFormatter = currencyFormatter;
+    this.accountCurrency = currencyFormatter.getCurrency();
+    this.decimalDigits = this.accountCurrency.decimal_digits;
+
+    // When 0, there are no currency sub-units => POS style entry is unneeded
+    return this.decimalDigits > 0;
+  }
+
+  invoke() {
+    this.wrapCurrencyInput('ynab-new-currency-input');
+  }
+
+  wrapCurrencyInput(name) {
+    const self = this;
+
+    const newCurrencyInputComponent = componentLookup(name);
+    const originalNewCurrencyInputComponentActions = Object.getPrototypeOf(
+      Object.getPrototypeOf(newCurrencyInputComponent)
+    ).actions;
```

Comment:
Couldn't make it work using Ember's [`reopen`](https://api.emberjs.com/ember/2.15/classes/Ember.Object/methods/reopen?anchor=reopen). Tried both with `_actions`/`actions` and on `newCurrencyInputComponent.constructor` and `newCurrencyInputComponent` proper.

Hadn't dug extremely deep, but there are some tricks/quirks/bugs to its use according to https://github.com/emberjs/ember.js/issues/5394